project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -108,6 +108,181 @@ function c4m_og_form_alter(&$form, $form_state) {
 }
 
 /**
+ * Implements hook_menu().
+ */
+function c4m_og_menu() {
+  $items['group/%/%/admin/owner'] = array(
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('c4m_og_replace_owner_build_form', 1, 2),
+    'access callback' => 'c4m_og_user_is_owner',
+    'access arguments' => array(1,2),
+    'title' => t('Replace Owner'),
+  );
+
+  return $items;
+}
+
+/**
+ * Form builder; OG manager change page.
+ */
+function c4m_og_replace_owner_build_form($form, $form_state, $entity_type, $etid) {
+  og_set_breadcrumb($entity_type, $etid, array(l(t('Group'), "$entity_type/$etid/group")));
+
+  $admins = array();
+  // Setting the list of the group admins.
+  foreach(c4m_og_get_group_admins($etid) as $admin) {
+    $admins[$admin->uid] = c4m_user_name($admin);
+  }
+
+  $form['new_owner'] = array(
+    '#type' => 'select',
+    '#options' => $admins,
+    '#title' => t('Select new owner'),
+    '#description' => t('Select the name of the member you wish to be the new owner.'),
+  );
+
+  $form['group'] = array(
+    '#type' => 'value',
+    '#value' => $etid,
+  );
+
+  $form['type'] = array(
+    '#type' => 'value',
+    '#value' => $entity_type,
+  );
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Replace'),
+  );
+
+  return $form;
+}
+
+/**
+ * Submit handler; Change the manager.
+ */
+function c4m_og_replace_owner_build_form_submit($form, &$form_state) {
+  $new_owner = array(
+    'uid' => $form_state['values']['new_owner'],
+    'name' => $form['new_owner']['#options'][$form_state['values']['new_owner']],
+  );
+  c4m_og_replace_owner($form_state['values']['type'], $form_state['values']['group'], $new_owner);
+
+  // Redirect back to the group page.
+  $form_state['redirect'] = $form_state['values']['type'] . '/' . $form_state['values']['group'] . '/group';
+}
+
+/**
+ * Validation function.
+ */
+function c4m_og_replace_owner_build_form_validate($form, &$form_state) {
+}
+
+/**
+ * Replacing the owner of a group.
+ *
+ * After replacing the owner of the group we grant "group administrator  member"
+ * role to the old owner and revoke this role from the new one.
```

Comment:
> The old GO becomes GA (admin role).

This is what is written in the [ticket](https://issuetracker.amplexor.com/jira/browse/CFM-1038) description.

And i'm removing the role from the new owner since it doesn't make any sense to keep it also for him.